### DISCUSSION TITLE
How to create framework in order to achieve this hierarchy?

### DISCUSSION THREAD

**Discussion by pranavirabatti** [Sunbird-Ed/Community#25] [Sunbird ED configuration discussion] [answered]
Expected: 
![image](https://user-images.githubusercontent.com/22250006/173815406-d7769c3f-0f9f-4a24-a4fc-39501b4c7001.png)

Existing:
![image](https://user-images.githubusercontent.com/22250006/173815463-bb9515a9-0cdf-4268-a44e-49d73ec5821f.png)

We have created the framework as follows:
https://pastebin.com/R3tUQbt0

Expected Values in the dropdown
Category:
Civil, Health, Accounts

Sub-Category:
Knowledge, Mission, Software

Language:
English, Hindi, Marathi

Category and Sub-category are not dynamic-dependent dropdowns.

The above-created framework is not working properly. What changes are to be done in the framework hierarchy to achieve the effect?

@vinukumar-vs

**Comment by luckynvdu** [comment]
@vinukumar-vs can you please respond? 

cc @rhwarrier

**Comment by vinukumar-vs** [comment]
[pranavirabatti](https://github.com/pranavirabatti) Where is this screen exists in the front-end? Mean in which workflow do we get the above-attached screens.

By updating the framework doesn't mean that it will appear on front-end screen. Some of front-end screen are driven by form-api & few by objectcategorydefination. Hence you have to update the same after framework API update.

You can verify the [framwork read API](http://docs.sunbird.org/latest/apis/framework/#operation/FrameworkV1ReadGet) is giving the expected response or not. If it is giving the expected response, then you have to update the right API(froms/objectcategorydefination) which is realted to the front-page.

Please share more details of the front-end page or in which workflow the attached screen will appear.
@sharathkashyap

**Reply by pramodkvarma** [reply]
@vinukumar-vs isn't this part of the first few pop up screens when user (without login or a cookie) shows up to ED portal where user is asked to choose framework for content suggestion?

**Reply by pramodkvarma** [reply]
Basically UX widgets showing framework selection is not generalized? @sharathkashyap @luckynvdu ?

**Reply by sharathkashyap** [reply]
@pramodkvarma We have not generalized the framework selection widgets yet.

**Comment by sharathkashyap** [comment, marked as answer]
@pranavirabatti The Framework should be set as provided in the curl below. I hereby attach the reference to framework defined.

`curl 'https://diksha.gov.in/api/framework/v1/read/ekstep_ncert_k-12?categories=board,gradeLevel,medium,class,subject' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-GB,en;q=0.9' \
  -H 'Connection: keep-alive' \
  -H 'Cookie: connect.sid=s%3Apcmfb4uYcQjIkkEu_y6fpdodxc_qRWSJ.cLjv6WOqjWQFjpSXvp%2Fkje4qHLs7Zg%2FBfhK0Pf3mdu0' \
  -H 'Referer: https://diksha.gov.in/explore?board=CBSE&gradeLevel=Class%2010&medium=English&selectedTab=textbook' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.115 Safari/537.36' \
  -H 'X-App-Id: prod.diksha.portal' \
  -H 'X-App-Version: 4.9.0' \
  -H 'X-Channel-Id: ORG_001' \
  -H 'X-Device-ID: b90d524477ee16107b666c254e39a19a' \
  -H 'X-Org-code: ORG_001' \
  -H 'X-Request-ID: aeb3dc39-310a-b6ac-b187-b3477911f91f' \
  -H 'X-Session-ID: 40ab295c-45a6-d872-3216-d242b3a48270' \
  -H 'X-Source: web' \
  -H 'X-msgid: aeb3dc39-310a-b6ac-b187-b3477911f91f' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="102", "Google Chrome";v="102"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'ts: 2022-06-28T09:27:07+05:30' \
  --compressed`

**Comment by luckynvdu** [comment]
@pranavirabatti Is your issue resolved? 
@sharathkashyap

**Reply by sharathkashyap** [reply]
@pranavirabatti We will close this thread due to inactivity
